components/match2/wikis/pokemon/match_summary.lua

```diff
@@ -0,0 +1,316 @@
+		local championBan = ChampionBan()
+
+		for gameIndex, banData in ipairs(championBanData) do
+			championBan:banRow(banData, gameIndex, banData.numberOfBans, match.date)
+		end
+
+		body:addRow(championBan)
+	end
+
+	return body
+end
+
+function CustomMatchSummary._createGame(game, gameIndex, date)
+	local row = MatchSummary.Row()
+	local extradata = game.extradata or {}
+
+	local championsData = {{}, {}}
+	local championsDataIsEmpty = true
+	for champIndex = 1, _NUM_CHAMPIONS_PICK do
+		if String.isNotEmpty(extradata['team1champion' .. champIndex]) then
+			championsData[1][champIndex] = extradata['team1champion' .. champIndex]
+			championsDataIsEmpty = false
+		end
+		if String.isNotEmpty(extradata['team2champion' .. champIndex]) then
+			championsData[2][champIndex] = extradata['team2champion' .. champIndex]
+			championsDataIsEmpty = false
+		end
+		championsData[1].color = extradata.team1side
+		championsData[2].color = extradata.team2side
+	end
+
+	if
+		Table.isEmpty(game.scores) and
+		String.isEmpty(game.winner) and
+		championsDataIsEmpty
+	then
+		return nil
+	end
+
+	row	:addClass('brkts-popup-body-game')
+		:css('font-size', '85%')
+		:css('overflow', 'hidden')
+
+	local score
+	if not Table.isEmpty(game.scores) then
+		score = table.concat(game.scores, '-')
+	end
+
+	row:addElement(CustomMatchSummary._opponentChampionsDisplay(championsData[1], _NUM_CHAMPIONS_PICK, date, false))
+	row:addElement(CustomMatchSummary._createCheckMark(game.winner == 1))
+	row:addElement(mw.html.create('div')
+		:addClass('brkts-popup-body-element-vertical-centered')
+		:wikitext(CustomMatchSummary._createAbbreviation{
+			title = String.isEmpty(game.length) and ('Game ' .. gameIndex .. ' picks') or 'Map Scores',
+			text = score or ('Game ' .. gameIndex),
+		})
+	)
+	row:addElement(CustomMatchSummary._createCheckMark(game.winner == 2))
+	row:addElement(CustomMatchSummary._opponentChampionsDisplay(championsData[2], _NUM_CHAMPIONS_PICK, date, true))
+
+	-- Add Comment
+	if not Logic.isEmpty(game.comment) then
+		row:addElement(MatchSummary.Break():create())
+		local comment = mw.html.create('div')
+		comment :wikitext(game.comment)
+				:css('margin', 'auto')
+		row:addElement(comment)
+	end
+
+	return row
+end
+
+function CustomMatchSummary._createCheckMark(isWinner)
+	local container = mw.html.create('div')
+		:addClass('brkts-popup-spaced')
+		:css('line-height', '27px')
+		:css('margin-left', '3%')
+		:css('margin-right', '3%')
+
+	if isWinner then
+		container:node(_GREEN_CHECK)
+	else
+		container:node(_NO_CHECK)
+	end
+
+	return container
+end
+
+function CustomMatchSummary._createAbbreviation(args)
+	return '<i><abbr title="' .. args.title .. '">' .. args.text .. '</abbr></i>'
+end
+
+function CustomMatchSummary._opponentChampionsDisplay(opponentChampionsData, numberOfChampions, date, flip, isBan)
+	local opponentChampionsDisplay = {}
+	local color = opponentChampionsData.color or ''
+	opponentChampionsData.color = nil
```

Comment:
```suggestion
	local color = Table.extract(opponentChampionsData.color) or ''
```

Reply:
```suggestion
	local color = Table.extract(opponentChampionsData, color) or ''
```

Reply:
@Hesketh2  ^ this would be the source of the error

Reply:
Im not sure if I skip any steps
![image](https://user-images.githubusercontent.com/88981446/182441168-1134419a-1dd4-4b28-885f-865b5410779d.png)

They are sitll returning the popup error but now the Lua error text indicates a nil table index, it's error seems to be stuck on these three line

272
226
136

I will update the revision into the git so that things can be matched here

Reply:
change `color` to `'color'`